cmd/discovery/main.go

```diff
@@ -1,53 +1,670 @@
-		fmt.Println(err)
-		os.Exit(1)
+		logger.Debugf("Warning: Could not load configuration: %v", err)
+		// Continue with defaults - not a fatal error
+	}
+
+	// Initialize AK Manager for transient AK approach
+	logger.Debugf("Initializing AK Manager for transient AK approach")
+	akManagerOpts := []tpm.Option{}
+	if config != nil && config.Config.Kcrypt.Challenger.TPMDevice != "" {
+		logger.Debugf("Using TPM device: %s", config.Config.Kcrypt.Challenger.TPMDevice)
+		akManagerOpts = append(akManagerOpts, tpm.WithTPMDevice(config.Config.Kcrypt.Challenger.TPMDevice))
+	}
+	akManager, err := tpm.NewAKManager(akManagerOpts...)
+	if err != nil {
+		return fmt.Errorf("creating AK manager: %w", err)
+	}
+	logger.Debugf("AK Manager initialized successfully")
+
+	// Get EK for attestation (transient AK approach)
+	logger.Debugf("Getting EK for attestation")
+	ek, err := akManager.GetEK()
+	if err != nil {
+		return fmt.Errorf("getting EK: %w", err)
+	}
+	logger.Debugf("Attestation data retrieved successfully")
+
+	// Compute TPM hash from EK using attestation helper (SHA-256 of EK SPKI)
+	logger.Debugf("Computing TPM hash from EK")
+	tpmHash, err := attpkg.ComputeTPMHashFromEK(ek)
+	if err != nil {
+		return fmt.Errorf("computing TPM hash: %w", err)
 	}
+	logger.Debugf("TPM hash computed successfully: %s", tpmHash)
+
+	// Output the TPM hash to stdout
+	fmt.Print(tpmHash)
+	return nil
 }
 
-// isEventDefined checks whether an event is defined in the bus.
-// It accepts strings or EventType, returns a boolean indicating that
-// the event was defined among the events emitted by the bus.
-func isEventDefined(i interface{}) bool {
-	checkEvent := func(e pluggable.EventType) bool {
-		if e == bus.EventDiscoveryPassword {
-			return true
+// runGetPassphrase handles the get subcommand - passphrase retrieval
+func runGetPassphrase(flags *GetFlags) error {
+	// Create logger based on debug flag
+	var logger types.KairosLogger
+	if debug {
+		logger = types.NewKairosLogger("kcrypt-discovery-challenger", "debug", false)
+	} else {
+		logger = types.NewKairosLogger("kcrypt-discovery-challenger", "error", false)
+	}
+
+	// Create client with potential CLI overrides
+	c, err := createClientWithOverrides(flags.ChallengerServer, flags.EnableMDNS, flags.ServerCertificate, logger)
+	if err != nil {
+		return fmt.Errorf("creating client: %w", err)
+	}
+
+	// Create partition object
+	partition := &block.Partition{
+		Name:            flags.PartitionName,
+		UUID:            flags.PartitionUUID,
+		FilesystemLabel: flags.PartitionLabel,
+	}
+
+	// Log partition information
+	logger.Debugf("Partition details:")
+	logger.Debugf("  Name: %s", partition.Name)
+	logger.Debugf("  UUID: %s", partition.UUID)
+	logger.Debugf("  Label: %s", partition.FilesystemLabel)
+	logger.Debugf("  Attempts: %d", flags.Attempts)
+
+	// Get the passphrase using the same backend logic as the plugin
+	fmt.Fprintf(os.Stderr, "Requesting passphrase for partition %s (UUID: %s, Label: %s)...\n",
```

Comment:
stderr seems the wrong channel for this as something may check if stderr is returning any data and doing logic based on that?